pod/perl5380delta.pod

```diff
@@ -0,0 +1,1658 @@
+equivalent. The only difference is that it does not and will never disable
+any optimisations in the regex engine. This may make it more unstable in the
+sense that it may be called more or less times in the future, however the
+number of times it executes will truly match how the regex engine functions.
+For example, certain types of optimisation are disabled when C<(?{ ... })> is
+included in a pattern, so that patterns which are O(N) in normal use become
+O(N*N) with a C<(?{ ... })> pattern in them. Switching to C<(*{ ... })> means
+the pattern will stay O(N).
+
+=head2 New C<class> Feature
+
+A new B<experimental> syntax is now available for defining object classes,
+where per-instance data is stored in "field" variables that behave like
+lexicals.
+
+    use feature 'class';
+
+    class Point
+    {
+        field $x;
+        field $y;
+
+        method zero { $x = $y = 0; }
+    }
+
+This is described in more detail in L<perlclass>.  Notes on the internals of
+its implementation and other related details can be found in L<perlclassguts>.
+
+This remains a new and experimental feature, and is very much still under
+development. It will be the subject of much further addition, refinement and
+alteration in future releases.  As it is experimental, it yields warnings in
+the C<experimental::class> category.  These can be silenced by a
+C<no warnings> statement.
+
+    use feature 'class';
+    no warnings 'experimental::class';
+
+=head2 REG_INF has been raised from 65,536 to 2,147,483,647
+
+Many regex quantifiers used to be limited to U16_MAX in the past, but are
+now limited to I32_MAX, thus it is now possible to write /(?:word){1000000}/
+for example.  Note that doing so may cause the regex engine to run longer
+and use more memory.
+
+=head2 New API functions optimize_optree and finalize_optree
+
+There are two new API functions for operating on optree fragments, ensuring
+you can invoke the required parts of the optree-generation process that might
+otherwise not get invoked (e.g. when creating a custom LOGOP).  To get access
+to these functions, you first need to set a C<#define> to opt-in to using
+these functions.
+
+  #define PERL_USE_VOLATILE_API
+
+These functions are closely tied to the internals of how the interpreter
+works, and could be altered or removed at any time if other internal changes
+make that necessary.
+
+=head1 Security
+
+=head1 Incompatible Changes
+
+=head2 readline() no longer clears the stream error flag
+
+C<readline()>, also spelled C<< <> >>, would clear the handle's error
+flag after an error occurred on the stream.
+
+The error flag is now only cleared when an error occurs when reading
+from the child process for glob() in F<miniperl>.  This allows it to
+correctly report errors from the child process on close().
+
+Since the error flag is no longer cleared calling close() on the
+stream may fail and if the stream was not explicitly closed, the
+implicit close of the stream may produce a warning.
```

Comment:
One additional behaviour that came out of this in BBCs is that the eof flag is also not being cleared (PerlIO_clearerr() cleared both flags), which is probably worth mentioning.

Reply:
I've added a sentence, but am limited in my writing my by understanding.  I wrote:

> The eof flag is also not cleared as often, which may cause disruption in code relying on the old behavior.

Reply:
See https://github.com/rjbs/perl5/pull/1